client/web/compose/src/components/PageBlocks/RecordListBase.vue

```diff
@@ -1046,6 +1082,7 @@ export default {
     if (!this.inlineEditing) {
       this.refreshBlock(this.refresh, false, true)
     }
+    this.recordsPerPage = this.options.perPage
```

Comment:
This should be done in prepRecordList, as the name suggests, that's where all the prep happens
As you can see you're still using options.perPage in the prep method